climada/entity/exposures/litpop/litpop.py

```diff
@@ -0,0 +1,1199 @@
+        # here, total_asset_val is Produced Capital "pc"
+        # no_land=True returns value w/o the mark-up of 24% for land value
+    if fin_mode == 'pc_land':
+        return(u_fin.world_bank_wealth_account(cntry_iso3a, reference_year,
+                                               no_land=False)[1])
+        # no_land=False returns pc value incl. the mark-up of 24% for land value
+    if fin_mode == 'norm':
+        return 1
+    # GDP based total values:
+    gdp_value = u_fin.gdp(cntry_iso3a, reference_year)[1]
+    if fin_mode == 'gdp':
+        return gdp_value
+    if fin_mode == 'income_group': # gdp * (income group + 1)
+        return gdp_value*(u_fin.income_group(cntry_iso3a, reference_year)[1]+1)
+    if fin_mode in ('nfw', 'tw'):
+        wealthtogdp_factor = u_fin.wealth2gdp(cntry_iso3a, fin_mode == 'nfw',
+                                              reference_year)[1]
+        if np.isnan(wealthtogdp_factor):
+            LOGGER.warning("Missing wealth-to-gdp factor for country %s.", cntry_iso3a)
+            LOGGER.warning("Using GDP instead as total value.")
+            return gdp_value
+        return gdp_value * wealthtogdp_factor
+    raise ValueError(f"Unsupported fin_mode: {fin_mode}")
+
+def reproject_input_data(data_array_list, meta_list,
+                        i_ref=0,
+                        target_res_arcsec=None,
+                        global_origins=(-180.0, 89.99999999999991),
+                        resampling=rasterio.warp.Resampling.bilinear,
+                        conserve=None):
+    """
+    Reprojects all arrays in data_arrays to a given resolution –
+    all based on the population data grid.
+
+    Parameters
+    ----------
+    data_array_list : list or array of numpy arrays containing numbers
+        Data to be reprojected, i.e. list containing N (min. 1) 2D-arrays.
+        The data with the reference grid used to define the global destination
+        grid should be first in the list, e.g., pop (GPW population data)
+        for LitPop.
+    meta_list : list of dicts
+        meta data dictionaries of data arrays in same order as data_array_list.
+        Required fields in each dict are 'dtype,', 'width', 'height', 'crs', 'transform'.
+        Example:
+            {'driver': 'GTiff',
+             'dtype': 'float32',
+             'nodata': 0,
+             'width': 2702,
+             'height': 1939,
+             'count': 1,
+             'crs': CRS.from_epsg(4326),
+             'transform': Affine(0.00833333333333333, 0.0, -18.175000000000068,
+                                 0.0, -0.00833333333333333, 43.79999999999993)}
+        The meta data with the reference grid used to define the global destination
+        grid should be first in the list, e.g., GPW population data for LitPop.
+    i_ref : int (optional)
+        Index/Position of data set used to define the reference grid.
+        The default is 0.
+    target_res_arcsec : int (optional)
```

Comment:
I will reply here in response to all your comments on `reproject_input_data()`:

The 'reference' grid is the one that is used to _align_ the target grid to globally, so that there is no offset between litpop grids for different countries (c.f. docstring of `climada.util.coordinates.align_raster_data()`.
By default, the GPW grid (`pop`) is used as this reference.

The target resolution, however, is set by the user and not necessarily equal to the resolution defined in the meta of the GPW grid. Therefore `meta_list(i_ref)` is ot equal to `meta_out`. I agree that it is confusing to call this a reference, therefore I rename the parameter `i_ref` to `i_align` throughout the function. Does this resolve some of teh confusion for you?